src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -136,7 +136,7 @@ private void SerializeMethodDebugInfo(IMethodBody bodyOpt, int methodRid, Standa
             SerializeStateMachineLocalScopes(bodyOpt, methodHandle);
 
             // delta doesn't need this information - we use information recorded by previous generation emit
-            if (Context.Module.CommonCompilation.Options.EnableEditAndContinue && !IsFullMetadata)
```

Comment:
We had tests that unfortunately tested the wrong behavior. I have updated the tests.